### PR TITLE
Remove unnecessary bloated session in credit mining tests

### PR DESCRIPTION
**Pull request by ardhipoetra**
This pull request removed bloated session launching in simple credit mining test.

### CODE REVIEW

**Comment by ardhipoetra** [comment]
retest this please

last run : [required checks passed](https://jenkins.tribler.org/job/GH_Tribler_PR/3748/)

**Comment by devos50** [comment]
@ardhipoetra don't forget to rebase this one. Please let me know when it's ready.

**Comment by ardhipoetra** [comment]
retest this please

last run : [required checks passed](https://jenkins.tribler.org/job/GH_Tribler_PR/3793/)

**Comment by ardhipoetra** [comment]
except the pylint error, I think it's ready, @devos50

File: Tribler/Test/Core/CreditMining/mock_creditmining.py [hunk 1]
```diff
@@ -3,8 +3,12 @@
 
 Written by Ardhi Putra Pratama H
 """
+from mock import MagicMock
```
Comment (devos50) [inline Tribler/Test/Core/CreditMining/mock_creditmining.py]: Please avoid using the `mock` framework unless you absolutely need it. Introducing this framework adds another dependencies and requires updating all testing machines. Instead, you can use `MockObject`.

File: Tribler/Test/Core/CreditMining/mock_creditmining.py [hunk 2]
```diff
@@ -122,7 +126,20 @@ class MockLtSession(object):
...
+        self.get_cm_max_torrents_per_source = sessconfig.get_cm_max_torrents_per_source
+        self.get_cm_max_torrents_active = sessconfig.get_cm_max_torrents_active
+        self.get_cm_source_interval = sessconfig.get_cm_source_interval
+        self.get_cm_swarm_interval = sessconfig.get_cm_swarm_interval
+        self.get_cm_share_mode_target = sessconfig.get_cm_share_mode_target
+        self.get_cm_tracker_interval = sessconfig.get_cm_tracker_interval
+        self.get_cm_logging_interval = sessconfig.get_cm_logging_interval
+        self.get_cm_sources = sessconfig.get_cm_sources
+
+        self.add_observer = lambda dummy_1, dummy_2, _: None
```
Comment (devos50) [inline Tribler/Test/Core/CreditMining/mock_creditmining.py]: You can ignore all arguments of a lambda function with `*_`

File: Tribler/Test/Core/CreditMining/mock_creditmining.py [hunk 3]
```diff
@@ -142,6 +159,12 @@ def shutdown(self):
         """
         pass
 
+    def get_download(self, x):
+        """
+        mocking function to get download id
```
Comment (devos50) [inline Tribler/Test/Core/CreditMining/mock_creditmining.py]: "mocked function to get a download"

File: Tribler/Test/Core/CreditMining/mock_creditmining.py (hunk 2 above)
Comment (ardhipoetra) [inline Tribler/Test/Core/CreditMining/mock_creditmining.py]: Thanks for the input, this is very useful!

**Comment by ardhipoetra** [comment]
retest this please

last run : [okay](https://jenkins.tribler.org/job/GH_Tribler_PR/3798/)

**Comment by ardhipoetra** [comment]
@devos50, how about this?

File: Tribler/Test/Core/CreditMining/mock_creditmining.py [hunk 4]
```diff
@@ -122,7 +126,26 @@ class MockLtSession(object):
...
+        self.lm = MockObject()
+        self.lm.ltmgr = MockObject()
+        self.lm.boosting_manager = MockObject()
+        self.lm.download_exists = MockObject()
+        self.lm.channelcast_db = MockObject()
+        self.lm.torrent_store = MockObject()
+
+        sessconfig = SessionConfigInterface()
+
+        self.get_cm_policy = lambda _: SeederRatioPolicy
```
Comment (devos50) [inline Tribler/Test/Core/CreditMining/mock_creditmining.py]: Where are you using these attributes?

Comment (devos50) [inline Tribler/Test/Core/CreditMining/mock_creditmining.py]: Oh whoops, I didn't see this is located inside the `MockLtSession` class.

**Comment by devos50** [comment]
@ardhipoetra please squash and I'll merge!

**Comment by ardhipoetra** [comment]
@devos50, there you go

**Pull request merged**